firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -427,9 +425,16 @@ private Query whereHelper(@NonNull FieldPath fieldPath, Operator op, Object valu
               .getUserDataReader()
               .parseQueryValue(value, op == Operator.IN || op == Operator.NOT_IN);
     }
-    Filter filter = FieldFilter.create(fieldPath.getInternalPath(), op, fieldValue);
+    FieldFilter filter = FieldFilter.create(fieldPath.getInternalPath(), op, fieldValue);
     validateNewFilter(filter);
-    return new Query(query.filter(filter), firestore);
+    return filter;
+  }
+
+  // TODO(ehsann): This method will become public API. Change visibility and add documentation.
```

Comment:
Nit: Maybe "query"/"orquery" if you want to work on this with others :)

Reply:
Done